packages/next/src/server/dev/next-dev-server.ts

```diff
@@ -675,7 +675,7 @@ export default class DevServer extends Server {
           fetchCacheKeyPrefix: this.nextConfig.experimental.fetchCacheKeyPrefix,
           isrFlushToDisk: this.nextConfig.experimental.isrFlushToDisk,
           maxMemoryCacheSize: this.nextConfig.experimental.isrMemoryCacheSize,
-          ppr: this.nextConfig.experimental.ppr === true,
+          ppr: this.renderOpts.experimental.ppr,
```

Comment:
also wondering why this is changing. At first glance to me it seems like a bug that everything else reads from `this.nextConfig.experimental` except for this property

Reply:
Because the enablement of PPR is dependant on the runtime + page kind, the `experimental` object was added to `RenderOpts` within app, enabling us to scope down the usage of renderOpts to specific page types. Ideally, we'd want to scope down access of each of the config options to each of the specific route implementaitons and runtimes.

In the most ideal world, we wouldn't pass around these huge bag-of-properties and instead rely on DI/IOC to isolate the configuration passing to the specific functions/classes/modules that require specific information from the configuration/render layers. That would solve the problem of PPR changes leaking into and breaking pages running inside `pages/`.